internal/controller/statusreport/statusreport_adapter.go

```diff
@@ -238,3 +235,108 @@ func (a *Adapter) findUntriggeredIntegrationTestFromStatus(integrationTestScenar
 	}
 	return ""
 }
+
+// ReportSnapshotStatus reports status of all integration tests into Pull Requests from component snapshot or group snapshot
+func (a *Adapter) ReportSnapshotStatus(ctx context.Context, testedSnapshot *applicationapiv1alpha1.Snapshot) error {
+
+	statuses, err := gitops.NewSnapshotIntegrationTestStatusesFromSnapshot(testedSnapshot)
+	if err != nil {
+		a.logger.Error(err, "failed to get test status annotations from snapshot",
+			"snapshot.Namespace", testedSnapshot.Namespace, "snapshot.Name", testedSnapshot.Name)
+		return err
+	}
+
+	integrationTestStatusDetails := statuses.GetStatuses()
+	if len(integrationTestStatusDetails) == 0 {
+		// no tests to report, skip
+		a.logger.Info("No test result to report to GitHub, skipping",
+			"snapshot.Namespace", testedSnapshot.Namespace, "snapshot.Name", testedSnapshot.Name)
+		return nil
+	}
+
+	// get the component snapshot list that include the git provider info the report will be reported to
+	destinationSnapshots := make([]*applicationapiv1alpha1.Snapshot, 0)
+	if gitops.IsComponentSnapshot(testedSnapshot) {
+		destinationSnapshots = append(destinationSnapshots, testedSnapshot)
+	} else if gitops.IsGroupSnapshot(testedSnapshot) {
+		// get component snapshots from group snapshot annation GroupSnapshotInfoAnnotation
+		destinationSnapshots, err = status.GetComponentSnapshotsFromGroupSnapshot(ctx, a.client, testedSnapshot)
+		if err != nil {
+			a.logger.Error(err, "failed to get component snapshots included in group snapshot",
+				"snapshot.NameSpace", testedSnapshot.Namespace, "snapshot.Name", testedSnapshot.Name)
+			return fmt.Errorf("failed to get component snapshots included in group snapshot %s/%s", testedSnapshot.Namespace, testedSnapshot.Name)
+		}
+	}
```

Comment:
Maybe we can extract this logic for getting destination Snapshots to a separate function to reduce the cognitive complexity for `ReportSnapshotStatus` (currently 20).

Reply:
Thanks. I will work on it.

Reply:
Updated.